core/src/com/unciv/models/translations/Translations.kt

```diff
@@ -139,6 +147,40 @@ class Translations : LinkedHashMap<String, TranslationEntry>(){
         return percentComplete
     }
 
+    private fun calculatePercentageForNationsFile(language: String, notTranslatedNations: Array<Nation>): Pair<Int, Int> {
+
+        val translationFileName = "jsons/Nations/Nations_$language.json"
+        val translationFile = Gdx.files.internal(translationFileName)
+        if (!translationFile.exists()) {
+            // calculate how many fields are missing
+            val allTranslatables = Nation::class.java.declaredFields.count {
+                it.type == String::class.java || it.type == ArrayList::class.java}
+            return Pair(0, allTranslatables*notTranslatedNations.size)
+        }
+
+        var translationsOfThisLanguage = 0
+        var allTranslations = 0
+
+        val translatedNations = JsonParser().getFromJson(emptyArray<Nation>().javaClass, translationFileName)
+
+        for (nation in notTranslatedNations)
+        {
+            val translatedNation = translatedNations.find { it.name == nation.name }
+
+            for (field in nation.javaClass.declaredFields.
+                    filter { it.type == String::class.java || it.type == ArrayList::class.java}) {
+                field.isAccessible = true
+                if (translatedNation != null && // we could exit *before* this loop, however, we need it here to count not translated fields
+                        (field.name in setOf("name", "startBias") || // skip fields which must not be translated
```

Comment:
Also cityStateType, preferredVictoryType, outerColor, innerColor etc.
I'm not sure if this correctly compares lists, by the way, because the != comparison between them is probably by-reference and not by-value, no?

Reply:
> cityStateType, preferredVictoryType, outerColor, innerColor etc.
All of those are not `String` or `ArrayList` types, so they are filtered a few lines before.
However, I forgot to skip the empty records from the original. Do the changes.

With regard to the comparison, that's what I found:
> `==` (`!=`) operator in Kotlin only compares the data or variables, whereas in Java or other languages `==` is generally used to compare the references. 
> `=== ` (`!==`) operator is used to compare the reference of two variable or object. It will only be true if both the objects or variables pointing to the same object.